### PR TITLE
[FINERACT-1751] Add savings account transaction request to Batch API

### PR DESCRIPTION
**Pull request by taskain7**
## Description

Adding the following request to Batch API:
 - get saving account
 - deposit
 - withdraw
 - holdAmount
 - releaseAmount

### CODE REVIEW

File: fineract-provider/src/main/java/org/apache/fineract/batch/command/CommandStrategyProvider.java [hunk 1]
```diff
@@ -128,7 +128,21 @@ private static void init() {
         commandStrategies.put(
                 CommandContext.resource("loans\\/external-id\\/" + UUID_PARAM_REGEX + OPTIONAL_QUERY_PARAM_REGEX).method(GET).build(),
                 "getLoanByExternalIdCommandStrategy");
+        commandStrategies.put(CommandContext.resource("savingsaccounts\\/" + NUMBER_REGEX + OPTIONAL_QUERY_PARAM_REGEX).method(GET).build(),
```
Comment (adamsaghy) [inline fineract-provider/src/main/java/org/apache/fineract/batch/command/CommandStrategyProvider.java:131]: You can use a generic command strategy to cover all these scenarios like:
`SavingsAccountTransactionCommandStrategy` and use 
```
        final Map<String, String> queryParameters = CommandStrategyUtils.getQueryParameters(relativeUrl);
        final String command = queryParameters.get("command");
```
to get the command query param

File: fineract-provider/src/main/java/org/apache/fineract/batch/command/CommandStrategyProvider.java [hunk 2]
```diff
@@ -128,7 +128,21 @@ private static void init() {
...
+        commandStrategies.put(
+                CommandContext.resource("savingsaccounts\\/" + NUMBER_REGEX + "\\/transactions\\?command=deposit").method(POST).build(),
+                "depositSavingsAccountCommandStrategy");
+        commandStrategies.put(
+                CommandContext.resource("savingsaccounts\\/" + NUMBER_REGEX + "\\/transactions\\?command=withdrawal").method(POST).build(),
+                "withdrawalSavingsAccountCommandStrategy");
+        commandStrategies.put(
+                CommandContext.resource("savingsaccounts\\/" + NUMBER_REGEX + "\\/transactions\\?command=holdAmount").method(POST).build(),
+                "holdAmountSavingsAccountCommandStrategy");
+        commandStrategies.put(CommandContext
```
Comment (adamsaghy) [inline fineract-provider/src/main/java/org/apache/fineract/batch/command/CommandStrategyProvider.java:137]: Similarly as above having a generic one probably would be beneficial:
`SavingsAccountAdjustTransactionCommandStrategy`

File: fineract-provider/src/main/java/org/apache/fineract/batch/command/CommandStrategyProvider.java (hunk 1 above)
Comment (taskain7) [reply fineract-provider/src/main/java/org/apache/fineract/batch/command/CommandStrategyProvider.java:131]: fixed

File: fineract-provider/src/main/java/org/apache/fineract/batch/command/CommandStrategyProvider.java (hunk 2 above)
Comment (taskain7) [reply fineract-provider/src/main/java/org/apache/fineract/batch/command/CommandStrategyProvider.java:137]: fixed

**Review by adamsaghy — APPROVED** [review]
LGTM

**Pull request merged**